magda-web-client/src/Components/Common/AddDatasetProgressMeter.tsx

```diff
@@ -59,16 +81,20 @@ function createStepUrl(datasetId, item: StepItem) {
 }
 
 const AddDatasetProgressMeter = (props: PropsType) => {
+    const isEdit =
```

Comment:
I have the following route in the router:
- /dataset/add ---  the landing page
- /dataset/list
- /dataset/add/urls
- /dataset/add/metadata
- /dataset/add/metadata/:datasetId/:step?
- /dataset/add/bulk
- /dataset/edit/:datasetId/:step?
I probably can change those to 
- /dataset/:action/urls
- /dataset/:action/metadata
- /dataset/:action/metadata/:datasetId/:step?
- /dataset/:action/bulk
- /dataset/:action/:datasetId/:step?
Except:
- /dataset/add
- /dataset/list
Otherwise, I probably have to dispatch the route to different component myself.
Is that what you want? Or you have a better suggestion?
Probably with that is we seem to have to allow some invalid URL: like /dataset/edit/bulk etc.

Reply:
@t83714 I've pushed a commit that illustrates what I mean, let me know what you think?

Reply:
To clarify, what bothers me about the approach of parsing the URL in components is that I'd rather keep inspecting the URL restricted to `<Route>` components wherever we can. This makes it a lot easier to check what needs to be updated if we ever want to move something URL-wise.

Reply:
Thanks a lot~ Looks good~ 👍